x-pack/plugins/lens/public/visualizations/xy/add_layer.tsx

```diff
@@ -71,6 +90,36 @@ export function AddLayerButton({
     };
   };
 
+  const dataPanel = ({
+    type,
+    label,
+    icon,
+    disabled,
+    toolTipContent,
+  }: typeof supportedLayers[0]) => {
+    return {
+      panel: AddLayerPanelType.layerVisualizationType,
+      toolTipContent,
+      disabled,
+      name: <span className="lnsLayerAddButtonLabel">{label}</span>,
+      className: 'lnsLayerAddButton',
+      icon: icon && <EuiIcon size="m" type={icon} />,
+      ['data-test-subj']: `lnsLayerAddButton-${type}`,
+    };
+  };
+
+  const horizontalOnly = isHorizontalChart(state.layers);
+
+  const availableVisTypes = visualizationTypes.filter(
+    (t) => isHorizontalSeries(t.id as SeriesType) === horizontalOnly
+  );
+
+  // to decide: should we shift the availableVisTypes array?
```

Comment:
I don't think we need to shift vis types, I prefer to be consistent with the list and just highlight (as you are already doing) the already-used layer.
I just don't understand exactly what the sorting order of the chart types (it reuses the same order as in the main chart switcher, but I still don't understand what order is: table first then by alphabetically by group name?)

Reply:
> I just don't understand exactly what the sorting order of the chart types (it reuses the same order as in the main chart switcher, but I still don't understand what order is: table first then by alphabetically by group name?)

FWIW, I agree with @marco's comment on the visualization type ordering. Historically speaking, I believe the current order we offer the visualization groups (tabular, bar, goal and single value, etc.) in the full visualization selectors was largely based on direction/input from product management a long time ago. The individual options within each group are then sorted alphabetically. In this smaller, abbreviated list of visualization types, it doesn't really make sense to continue order them this way without their group headers. 

That said, I suggest we either 1) restore the group headers (bar, line and area) if we wish to attempt some consistency with the full visualization type selector, or 2) apply a simple alphabetical order. My immediate instinct is to say let's do the first suggestion in the short term, and then change to the second suggestion after we get around to simplifying the list of visualization types in the future. Thoughts?

Reply:
Good for me going with the first suggestion first and then we reiterate this later when chart types are simplified

Reply:
I am not sure if it's worth to think about it for long as we're gonna simplify the types, but I'd skip the group headers if that's ok. The reason is that the 'select visualization type' subpanel is longer than the main panel, and in some rare cases it shows the scroll on main lens page and we'd like to avoid it. Making it longer makes it happen more often.

![Apr-02-2024 13-30-11](https://github.com/elastic/kibana/assets/4283304/78af5289-9895-4589-a4a4-8f23e00b88ac)

Reply:
Fine for me to skip it for now